tools/generate_tables_sql_and_testNames.php

```diff
@@ -1,118 +1,234 @@
+                    $bits[0] = "varchar(255)";
+                } else if ($bits[0] == "checkbox") {
+                    $bits[0] = "varchar(255)";
+                } else if ($bits[0] == "static") {
+                    $bits[0] = "varchar(255)";
+                } else {
+                    // Sanitize $bits to mitigate SQL injection attacks.
+                    $bits[0] = str_replace('`', '', $db->escape($bits[0]));
                 }
-
-                $bits[2]=htmlspecialchars($bits[2]);
-                $output.="`$bits[1]` $bits[0] default NULL,\n";
+                $bits[2]     = htmlspecialchars($bits[2]);
+                $output     .= "`$sql_query_table_name` $bits[0] default NULL,\n";
+                $column_name = str_replace('`', '', $db->escape($bits[1]));
+                $sql_query_columns[] = "`$column_name` $bits[0] default NULL,\n";
+            }
         }
-
+        $output .= "PRIMARY KEY  (`CommentID`)\n) ENGINE=InnoDB"
+            ." DEFAULT CHARSET=utf8;\n";
+        $fp      = fopen($file_name, "w");
+        fwrite($fp, $output);
+        // Check if Table Name exists
+        if (empty($sql_query_table_name)) {
+            $error_message = "Table name does not exist";
+        } else {
+            fwrite(
+                $fp,
+                "REPLACE INTO test_names (Test_name, Full_name, Sub_group) "
+                ."VALUES ('$sql_query_table_name', '$title', 1);\n"
+            );
+            // Writes a record of the changes to a file but is not sourced directly.
+            foreach ($pages as $page_number => $page) {
+                fwrite(
+                    $fp,
+                    "INSERT INTO instrument_subtests "
+                    ."(Test_name, Subtest_name, Description)"
+                    ." VALUES ('$sql_query_table_name', '"
+                    .$sql_query_table_name."_page".($page_number+1)
+                    ."', '$page');\n"
+                );
+            }
+        }
+        // Writes the error message to the sql file.
+        if (!empty($error_message)) {
+            fwrite(
+                $fp,
+                "/* The SQL file is invalid! \n"
+                ."tools/generate_tables_sql_and_testNames.php error_message:\n"
+                ."$error_message\n"
+                ."*/"
+            );
+        }
+        fclose($fp);
     }
-    $output.="PRIMARY KEY  (`CommentID`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8;\n";
-    print "Filename: $filename\n";
-    $fp=fopen($filename, "w");
-    fwrite($fp, $output);
-    fwrite($fp, "REPLACE INTO test_names (Test_name, Full_name, Sub_group) VALUES ('$tablename', '$title', 1);\n");
-    foreach($pages as $pageNo => $page) {
-        fwrite($fp, "INSERT INTO instrument_subtests (Test_name, Subtest_name, Description) VALUES ('$tablename', '" . $tablename  . "_page" . ($pageNo+1) . "', '$page');\n");
-
+    // Execute the SQL tasks if no error_message and CLI flag set.
+    if ($db->isConnected() 
```

Comment:
not sure why you are checking if connected, we never check for that

Reply:
I used code I previously saw in LORIS when writing this and it was awhile ago. It should be removed.